stdlib/grpc/src/main/java/org/ballerinalang/net/grpc/ServicesBuilderUtils.java

```diff
@@ -69,16 +69,32 @@ public static ServerServiceDefinition getServiceDefinition(BRuntime runtime, Obj
         if (fileDescriptor == null) {
             throw new GrpcServerException("Couldn't find the service descriptor.");
         }
-        String serviceTypeName = service.getType().getName(); // typeName format: <name>$$<type>$$<version>
-        String[] splitValues = serviceTypeName.split("\\$\\$");
-        String serviceName = splitValues[0];
+        String serviceName = getServiceName(service);
         Descriptors.ServiceDescriptor serviceDescriptor = fileDescriptor.findServiceByName(serviceName);
         if (serviceDescriptor == null) {
             throw new GrpcServerException("Couldn't find the service descriptor for the service: " + serviceName);
         }
         return getServiceDefinition(runtime, service, serviceDescriptor);
     }
-    
+
+    private static String getServiceName(ObjectValue service) {
+        String serviceName = null;
+        Object serviceConfigData = service.getType().getAnnotation("ballerina/grpc:ServiceConfig");
+        if (serviceConfigData != null) {
+            MapValue<String, Object> configMap = (MapValue) serviceConfigData;
+            String providedName = configMap.getStringValue("name");
+            if (providedName != null && !providedName.isEmpty()) {
+                serviceName = providedName;
+            }
+        }
+        if (serviceName == null) {
```

Comment:
If we return the providedName from L87, we would not need this null check. WDYT?
If so, serviceName variable can be also removed